### DISCUSSION TITLE
Why do custom identifiers have such strict validation rules?

### DISCUSSION THREAD

**Discussion by kevinkucharczyk** [vanilla-extract-css/vanilla-extract#1176] [Q&A]
Recently, custom identifier hashing was added via https://github.com/vanilla-extract-css/vanilla-extract/pull/1160

However, the strings that are allowed for custom identifier functions are too restrictive. There is a regex match check here: https://github.com/vanilla-extract-css/vanilla-extract/blob/58005eb5e7456cf2b3c04ea7aef29677db37cc3c/packages/css/src/identifier.ts#L78, which only allows underscores, letters and numbers.

At the very least I'd like to be able to use hyphens, or escaped period characters, but CSS identifiers allow for more than that: https://stackoverflow.com/a/449000

Can the regex in custom identifiers be expanded to allow more characters? Is there a reason why this check is so strict?

**Comment by Yaojian** [comment]
@SombreroElGringo Is there a reason why the hyphen (`-`) cannot returned by the identifier function?

**Reply by SombreroElGringo** [reply]
@Yaojian no specific reason, I just re-implemented the following change who has been previously discussed/agreed in this [PR](https://github.com/vanilla-extract-css/vanilla-extract/pull/657#pullrequestreview-948182814), but I totally agree with you I think we should have more flexibility on the identifier.

**Reply by mrm007** [reply]
The hyphen is now allowed: https://github.com/vanilla-extract-css/vanilla-extract/pull/1292

If you'd like even more flexibility, feel free to open a PR.
